storage/buffer.go

```diff
@@ -358,30 +358,30 @@ func (it *sampleRingIterator) Next() chunkenc.ValueType {
 	}
 }
 
-func (it *sampleRingIterator) Seek(int64) chunkenc.ValueType {
+func (it *SampleRingIterator) Seek(int64) chunkenc.ValueType {
 	return chunkenc.ValNone
 }
 
-func (it *sampleRingIterator) Err() error {
+func (it *SampleRingIterator) Err() error {
 	return nil
 }
 
-func (it *sampleRingIterator) At() (int64, float64) {
+func (it *SampleRingIterator) At() (int64, float64) {
 	return it.t, it.f
 }
 
-func (it *sampleRingIterator) AtHistogram() (int64, *histogram.Histogram) {
+func (it *SampleRingIterator) AtHistogram() (int64, *histogram.Histogram) {
 	return it.t, it.h
 }
 
-func (it *sampleRingIterator) AtFloatHistogram() (int64, *histogram.FloatHistogram) {
+func (it *SampleRingIterator) AtFloatHistogram(fh *histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {
```

Comment:
Now that `SampleRingIterator` doesn't implement `chunkenc.Iterator` anymore, it would be good to have doc comments. At least for this method, that now has an additional parameter.

Reply:
I've added comments on all methods. I also removed `Seek` and `Err` because they were never called. Also `Seek` was poorly implemented and always returned `chunkenc.ValNone`. Let me know if I should revert this change.

Reply:
Yead, Seek was only implemented to implement the `chunkenc.Iterator` interface. But as proven by your change, we don't need that.